### DISCUSSION TITLE
Z_cluster id changes after running Link job

### DISCUSSION THREAD

**Discussion by iitbhumanish** [zinggAI/zingg#600] [Q&A] [answered]
Hi ,

Just need quick information after running Link job.  I see that Z_Cluster id is changed after link job , it did not use same Cluster id  created while running Match job for the same match records.

**Is there a way to get LINK job to use same cluster id generated through  MATCH job ? Please advise.**

For ex:  four duplicate records Stewart were kept under Cluster Id = 8305 after MATCH job 
![image](https://github.com/zinggAI/zingg/assets/131442577/42aef4f8-d07c-4202-adff-eb76ec1a1ba2)

As part of Incremental load , it generates different cluster id for another incoming duplicate Stewart after LINK job .

![image](https://github.com/zinggAI/zingg/assets/131442577/e0fbb131-580e-4abf-b830-68a391b6bdd3)

**Comment by sonalgoyal** [comment]
can you please tell me what are the input datasets you are passing to the link phase?

**Reply by iitbhumanish** [reply]
I am passing both Exist and Delta as input. Please refer below screen shots

Note : For existing record,  I had converted matched records ( generated through MATCH job)  into temp file without three extra fields : minScore, maxScore and ClusterId

![image](https://github.com/zinggAI/zingg/assets/131442577/6b5f6c16-4c96-40a3-a345-b59fccdcb5e4)

**Comment by sonalgoyal** [comment, marked as answer]
Got it, thanks. In Zingg OSS, link is a different phase from match and designed to find record similarity across datasets and not designed for incremental processing post a one time match. Hence you see the results you are getting.

**Reply by iitbhumanish** [reply]
Thanks Sonal for the response.
